secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/dcr/request/DCRSsaSignatureValidator.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.dcr.request;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.SignatureException;
+
+import javax.validation.constraints.NotNull;
```

Comment:
Why is the Java Bean Validation API being used? 

This API is typically aimed at validating user input, it is used in the Open Banking datamodel as part of the generated code. Spring can be configured to process these annotations and apply the validation.

IG will not be processing these annotations, so the validation will not be being applied.

Reply:
Ah, yes. I was thinking at one stage that IG did include a validator, but perhaps it didn't I sometimes find @NotNull good as a documentation mechanism, and of course we could always choose to put the hibernate validator on the class path and initialise the validator. 

However, I agree that I started off using @NotNull, but it's usage is inconsistent, so probably better off removed.